src/main/scala/strawman/collection/immutable/Vector.scala

```diff
@@ -0,0 +1,1140 @@
+
+  override def last: A = {
+    if (isEmpty) throw new UnsupportedOperationException("empty.last")
+    apply(length - 1)
+  }
+
+  /*override*/ def init: Vector[A] = {
+    if (isEmpty) throw new UnsupportedOperationException("empty.init")
+    dropRight(1)
+  }
+
+  override def slice(from: Int, until: Int): Vector[A] =
+    take(until).drop(from)
+
+  override def splitAt(n: Int): (Vector[A], Vector[A]) = (take(n), drop(n))
+
+  // concat (suboptimal but avoids worst performance gotchas)
+//  override def concat[B >: A](that: IterableOnce[B]): Vector[B] = {
+//    import Vector.{Log2ConcatFaster, TinyAppendFaster}
+//    if (that.isEmpty) this
+//    else {
+//      that.size match {
+//        // Often it's better to append small numbers of elements (or prepend if RHS is a vector)
+//        case n if n <= TinyAppendFaster || n < (this.size >>> Log2ConcatFaster) =>
+//          var v: Vector[B] = this
+//          for (x <- that) v = v :+ x
+//          v
+//        case n if this.size < (n >>> Log2ConcatFaster) && that.isInstanceOf[Vector[_]] =>
+//          var v = that.asInstanceOf[Vector[B]]
+//          val ri = this.reverseIterator
+//          while (ri.hasNext) v = ri.next +: v
+//          v
+//        case _ => super.concat(that)
+//      }
+//    }
+//  }
+
+  // semi-private api
+
+  private[immutable] def updateAt[B >: A](index: Int, elem: B): Vector[B] = {
+    val idx = checkRangeConvert(index)
+    val s = new Vector[B](startIndex, endIndex, idx)
+    s.initFrom(this)
+    s.dirty = dirty
+    s.gotoPosWritable(focus, idx, focus ^ idx)  // if dirty commit changes; go to new pos and prepare for writing
+    s.display0(idx & 31) = elem.asInstanceOf[AnyRef]
+    s
+  }
+
+  private def gotoPosWritable(oldIndex: Int, newIndex: Int, xor: Int) = if (dirty) {
+    gotoPosWritable1(oldIndex, newIndex, xor)
+  } else {
+    gotoPosWritable0(newIndex, xor)
+    dirty = true
+  }
+
+  private def gotoFreshPosWritable(oldIndex: Int, newIndex: Int, xor: Int) = if (dirty) {
+    gotoFreshPosWritable1(oldIndex, newIndex, xor)
+  } else {
+    gotoFreshPosWritable0(oldIndex, newIndex, xor)
+    dirty = true
+  }
+
+  /** A copy of the $coll with an element prepended.
+   *
+   *  @param  value  the prepended element
+   *  @tparam B      the element type of the returned $coll.
+   *  @return a new collection of type `Vector[B]` consisting of `value` followed
+   *          by all elements of this $coll.
+   *
+   *    @inheritdoc
+   *
+   *    Also, the original $coll is not modified, so you will want to capture the result.
+   *
+   *    Example:
+   *    {{{
+   *      scala> val x = List(1)
+   *      x: List[Int] = List(1)
+   *
+   *      scala> val y = 2 +: x
+   *      y: List[Int] = List(2, 1)
+   *
+   *      scala> println(x)
+   *      List(1)
+   *    }}}
+   *
+   *    @return a new $coll consisting of `value` followed
+   *            by all elements of this $coll.
+   */
+  def appendFront[B >: A](value: B): Vector[B] = {
```

Comment:
I kept the name `appendFront` (and `appendBack`, below) from the current implementation of `Vector`. These methods were private and I made them public because we need an alphabetic alias for `+:` and `:+`. Are you ok with these names or do you prefer `prepend` and `append`?